### ISSUE TITLE
Rule proposal: `no-named-default`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2087]
### Description

During refactoring, eg:

```js
// We don't need `foo` at first.
doThingsWithoutFoo();

export {default as foo} from 'foo';
```

```js
import {default as foo} from 'foo'; // Didn't mean to import like this, but changed from previous version

// We need `foo` now
doThingsWithFoo();
export {foo}
```

### Fail

```js
import {default as foo} from 'foo';
```

```js
import {default as foo, bar} from 'foo';
```

### Pass

```js
import foo from 'foo';
```

```js
import foo, {bar} from 'foo';
```

### Additional Info

Should we check export?

#### Fail

```js
export {foo as default};
```

#### Pass

```js
export default foo;
```

**Comment by fisker** [comment]
But it doesn't have an auto-fix, let's make our own?

**Comment by sindresorhus** [comment]
> Should we check export?

Yes

> But it doesn't have an auto-fix, let's make our own?

👍

**Comment by fisker** [comment]
For  implemention details:

This case

```js
import foo, {default as bar} from 'foo';
```

Should we report? Maybe user have a good reason to do that. 

```js
import foo, {default as bar} from 'foo';

function a({foo}) {
   // For a good reason that he can't use `foo` here.
   bar(...)
}
```

The following seems not an improvement.

```js
import foo from 'foo';
const bar = foo;

function a({foo}) {
   // For a good reason that he can't use `foo` here.
   bar(...)
}
```

Maybe we should report and ask user to use `eslint-disable`?

**Comment by sindresorhus** [comment]
> Should we report? Maybe user have a good reason to do that.

Yes. We should report. That's just a confusing pattern. Variable reassignment (`const bar = foo;`) is much clearer.

**Comment by fisker** [comment]
Okay, report but not fixable.

**Comment by B2o5T** [comment]
> @B2o5T You don't like the idea, or don't like that we are making a new version of import/no-named-default. Can you share your thought?

I don't like the idea because there is already an existing rule.

note that similar extending existing rule with auto fix [no-negated-condition](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-negated-condition.md) is relevant due to big refactoring in codebases

**Comment by B2o5T** [comment]
> I don't like to do this either, but I like this rule to have auto-fix.
> 
> Are you suggestion to contribute to `eslint-plugin-import` instead?

You can try to propose implementing autofix in eslint-plugin-import

**Comment by fisker** [comment]
Main reason I didn't try that.

> That plugin is not actively maintained, the rule only covers some of what's mentioned here, and the plugin is slow as it traverses all the imports and their graph.

Quote from https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1067#issuecomment-767022653, and I agree with that.

I sent other PRs there before and the develop experience are bad, as they even need support [Node.js  v4](https://github.com/import-js/eslint-plugin-import/blob/328064abc707d3289772a8a29da5783c6dc345f6/.github/workflows/node-4%2B.yml#L96) and [ESLint v2](https://github.com/import-js/eslint-plugin-import/blob/328064abc707d3289772a8a29da5783c6dc345f6/.github/workflows/node-4%2B.yml#L38)

**Comment by sindresorhus** [comment]
Accepted

**Comment by fisker** [comment]
We decide to also check `export {foo as default}`, I guess rename to `no-named-default` is reasonable.

**Comment by sindresorhus** [comment]
> Maybe no-default-export-as-named is a better name?

I think `no-named-default` is just as clear and shorter. And the above misses `import`, so must have been `no-default-export-import-as-named`.
